src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -741,9 +741,18 @@ protected override void Normalize(ref LocalState state)
             {
                 var id = variableBySlot[i];
                 int slot = id.ContainingSlot;
-                state.Assigned[i] = (slot > 0) &&
+
+                bool assign = (slot > 0) &&
                     state.Assigned[slot] &&
                     variableBySlot[slot].Symbol.GetTypeOrReturnType().TypeKind == TypeKind.Struct;
+
+                if (state.NormalizeToBottom)
+                {
+                    // NormalizeToBottom means new variables are assumed to be assigned (bottom state)
+                    assign |= slot == 0;
```

Comment:
>slot == 0 [](start = 30, length = 9)

Could we move this into the ```if``` condition and use ```assign = true``` here? I think that would be easier to follow.